src/providers/sn_provider/starknet_provider.rs

```diff
@@ -38,25 +38,55 @@ where
         Self { provider }
     }
 
-    /// Retrieves the balance of a Starknet address for a specified block.
+    /// Retrieves the balance of the parent token for a Starknet address.
     ///
-    /// This method interacts with the Starknet native token contract to query the balance of the given
-    /// address at a specific block.
+    /// # Arguments
+    /// * `token_address` - The address of the token contract
+    /// * `address` - The address to check the balance for
+    /// * `block_id` - The block ID at which to check the balance
     ///
-    /// If the contract is not deployed or the class hash is not declared, a balance of 0 is returned
-    /// instead of an error.
-    pub async fn balance_at(&self, address: Felt, block_id: BlockId) -> Result<U256, ExecutionError> {
-        // Create a new `ERC20Reader` instance for the Starknet native token
-        let eth_contract = ERC20Reader::new(*STARKNET_NATIVE_TOKEN, &self.provider);
+    /// # Returns
+    /// * `Result<U256, ExecutionError>` - The balance as a U256 value or an execution error
+    pub async fn balance_at_parent(&self, address: Felt, block_id: BlockId) -> Result<U256, ExecutionError> {
```

Comment:
i don't think we should call ETH parent in terms of naming

We could make in unambiguous by just calling it: eth_balance_at and strk_balance_at

Reply:
Ok wait nevermind 
you are implying that since this codebase can be used by any starknet appchain
some appchains might want to use their own token as base token

Reply:
at any rate, since STARKNET_PARENT_TOKEN is a fixed address and not even an injected variable. For now it only ever points to ETH
So I think my point remains valid

If one day we want to parametrize base token, we will do it